ui/desktop/tests/acceptance/projects/sessions/session-test.js

```diff
@@ -148,13 +148,73 @@ module('Acceptance | projects | sessions | session', function (hooks) {
     await click(TARGET_CONNECT_BUTTON);
 
     assert.strictEqual(currentURL(), urls.session);
-    assert.dom('credential-panel-header').doesNotExist();
-    assert.dom('credential-panel-body').doesNotExist();
+    assert.dom('.credential-panel-header').doesNotExist();
+    assert.dom('.credential-panel-body').doesNotExist();
     assert
       .dom('[data-test-no-credentials]')
       .hasText(`Connected You can now access ${instances.target.name}`);
   });
 
+  test('visiting session with credentials', async function (assert) {
+    this.ipcStub.withArgs('cliExists').returns(true);
+    this.ipcStub.withArgs('connect').returns({
+      session_id: instances.session.id,
+      address: 'a_123',
+      port: 'p_123',
+      protocol: 'tcp',
+      credentials: [
+        {
+          credential_source: {
+            id: 'clvlt_4cvscMTl0N',
+            name: 'Credential Library 0',
+            description: 'Source Description',
+            credential_store_id: 'csvlt_Q1HFGt7Jpm',
+            type: 'vault-generic',
+          },
+          secret: {
+            raw: 'eyJhcnJheSI6WyJvbmUiLCJ0d28iLCJ0aHJlZSIsIm9uZSIsInR3byIsInRocmVlIiwib25lIiwidHdvIiwidGhyZWUiLCJvbmUiLCJ0d28iLCJ0aHJlZSJdLCJuZXN0ZWQiOnsiYm9vbCI6dHJ1ZSwibG9uZyI6IjEyMjM1MzQ1NmFzZWRmYTQzd3J0ZjIzNGYyM2FzZGdmYXNkZnJnYXdzZWZhd3NlZnNkZjQiLCJzZWNlcmV0Ijoic28gbmVzdGVkIn0sInRlc3QiOiJwaHJhc2UifQ',
+            decoded: {
+              data: {
+                backslash: 'password\\withslash\\',
+                backslash1: 'password\\fwithslash\\f',
+                password: '123',
+                username: 'test',
+                email: {
+                  address: 'test.com',
+                },
+              },
+              metadata: {
+                created_time: '2024-04-12T18:38:36.226715555Z',
+                custom_metadata: null,
+                deletion_time: '',
+                destroyed: false,
+                version: 8,
+              },
+            },
+          },
+        },
+      ],
+    });
+
+    await visit(urls.target);
+    await click(TARGET_CONNECT_BUTTON);
+    assert.strictEqual(currentURL(), urls.session);
+
+    assert.dom('.credential-panel-header').exists();
+    assert.dom('.credential-panel-body').exists();
+    assert.dom('.credential-secret').exists();
+
+    assert
+      .dom('.secret-container:nth-of-type(1)')
+      .hasText('username ■ ■ ■ ■ ■ ■ ■ ■ ■ ■ ■ ■');
+
+    await click('.secret-container:nth-of-type(1) .hds-icon');
+
+    assert
+      .dom('.secret-container:nth-of-type(1) .secret-content')
+      .hasText('test');
```

Comment:
Shouldn't we also be checking/asserting the fields with escaped characters? Actually, did we also have a test for this for vault credential as well?

Reply:
testing this is quite tricky- dom assertions output contain text as `password\\withslash\\` (with esc chars) but the removal of esc happens when it is displayed on the screen, that's why I didn't include esc chars test. is there a way to test this? 🤔 

`assert.dom('.secret-container:nth-of-type(3) .secret-content').hasText("password\\withslash\\");` 
This is as same as the input we pass to this assert, which to me didn't seem to add any value. I also tried asserting innerHTML, which also shows the esc chars.

Reply:
> Actually, did we also have a test for this for vault credential as well?

this test is for vault credential, did you mean one for static? I added one for static now

Reply:
> this test is for vault credential, did you mean one for static? I added one for static now

I meant for the previous PR that handled the vault credential, did we also have a test for escape characters?

I would use template strings to check against (which will display it raw with no escaping) and compare that to what's being displayed. Using normal strings will always have to have escape characters.

This is also something we should add an E2E test for (not for this PR) but this is something we definitely should validate and getting in the habit of adding

Reply:
agreed that was my plan to handle this in e2e, added a [ticket](https://hashicorp.atlassian.net/browse/ICU-15986) for e2e test.

Reply:
I can add more details and link slack conversation to this ticket in a bit, can you give me a link to Admin UI where we show this credential panel? As far as I'm aware, this is solely a desktop change, so I'm wondering if I missed this 

this is a parent [ticket](https://hashicorp.atlassian.net/browse/ICU-13856) where the issue is reported

Reply:
We have no showing credentials panel within Admin, but following the [new ticket created](https://hashicorp.atlassian.net/browse/ICU-15986) what I mean is:
- The ticket created is under Platform Maintenance, wich is a none scoped parent. In other words, can apply to anything within the code base.
- The ticket created title is: "e2e test for credentials display" and has no description. So someone, can missunderstood it as add an e2e test for the credentials store screen within Admin.

I am not sure we discuss anything in slack, but within this PR comment.

Reply:
ah ok I thought you were talking about this PR's ticket. 

I updated the e2e ticket and added further context and linked parent issue

Reply:
@ZedLi I added tests for both static and vault that handle nested data and esc chars, thanks for the feedback on using template strings.